hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestInsertTable.scala

```diff
@@ -1046,4 +1048,62 @@ class TestInsertTable extends HoodieSparkSqlTestBase {
       )
     }
   }
+
+  /**
+   * This test is to make sure that bulk insert doesn't create a bunch of tiny files if
+   * hoodie.bulkinsert.user.defined.partitioner.sort.columns doesn't start with the partition columns
+   *
+   */
+  forAll(BulkInsertSortMode.values().toList) { (sortMode: BulkInsertSortMode) =>
+    val sortModeName = sortMode.name()
+    test(s"Test Bulk Insert with BulkInsertSortMode: '$sortModeName'") {
```

Comment:
Unfortunately, i don't think this is going to work -- Scalatest requires `test` method have to be at the top level, so we'd have to move this iteration into it

Reply:
I did this when I made TestSparkSqlCoreFlow and it seems to work. It is a bit annoying that you need to comment out the other tests in the class if you just want to run this test, but when we made the core flow test, it seemed to be the best way to do this

Reply:
Oh, nice! Didn't know that. Previously i've tried just using a normal for-loop and that was just shamble